### ISSUE TITLE
WebUI - Camera Viewer control

### ISSUE THREAD

**Issue by VaughnRupp** [opensourceautomation/Open-Source-Automation#237]
We will need to create a camera viewer control for the webui that approxamates what is in the Screens app for consistency of rendering user's screens across tech.

**Comment by KHerron** [comment]
This should be fixed with my last pull request

**Comment by VaughnRupp** [comment]
I tried by taking a snapshot everytime the screen refreshes.   it works with my Loftek, but not with my Foscam.   So this went out in 048, but I am pushing the issue to 049 until it works for both my cameras or we get an actual stream working in the control.

Kirk, you are welcome to assign yourself to this one, if not, I will eventually get back around to it.   I will leave it unassigned for now.

**Comment by KHerron** [comment]
I can’t assign myself on git. I think only you and Brian can.

But I will be happy to look in to it.

 

Will be updating my system over the weekend to your latest commits.

I plan to go thru all my plugins, and get them current and updated, so I will be work on that too!

I also want to get my new user controls working and updated to the latest, and get them published to the website.

 

 

Kirk Herron 
Project Manager 
(904)626-9854 
CCSG, Inc.

ES12000916 
http://www.ccsgweb.com <http://www.ccsgweb.com/>

**Comment by VaughnRupp** [comment]
I guess I missed the message
"KHerron  commented on Nov 17 2015 " or I would not have hacked up what was on the ASP pages for this.   You are welcome to change it back or to whatever control you think best and I will test it with my 2 cameras.    I would like to get this control stable on the web for 049.

**Comment by KHerron** [comment]
No, if it didn't work for you then it probably wont work at all. I never did test it fully, just figured that since I could get to a You tube video, that it would have worker. I have left your changes alone, and have created a new VideoStreamViewer WebUI control, to play with to see if I can solve this issue. The biggest problem I am having is "Cross-Browser" support. Chome has really changed it's streaming functions and abilities. However, with that said, I do have one working right now, but ONLY in chrome. It fails to open the stream in IE. Also I don't have the latest IE, Firefox or Safari to test it with. I will have to add some Browser specific code. I can submit my changes so you can test it if you want. All I did is changed the screens.aspx.cs to loadt the new control instead of the ctrlEmbedded. This way not much was changed, and can always be changed back. The MjpegProcessor will not run on the website, so I can't try to re-use that.

**Comment by VaughnRupp** [comment]
I did not get to read the whole thing, and it looks like a Windows app, but he does show a web page with a stream on it.   Either way, it is well written and I ready the end part of this, so I though I would share incase you hadn't seen it and maybe can get something from it...
http://www.codeproject.com/Articles/15537/Camera-Vision-video-surveillance-on-C
